sound/soc/sof/intel/hda-dsp.c

```diff
@@ -467,26 +467,35 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 
 		if (hda_dsp_d0i3_playback_active(sdev))
 			flags |= HDA_PM_PG_STREAMING;
+
+		ret = hda_dsp_send_pm_gate_ipc(sdev, flags);
+		if (ret < 0) {
+			dev_err(sdev->dev, "PM_GATE ipc error %d\n", ret);
+			return ret;
+		}
+
+		/* update D0I3C register */
+		ret = hda_dsp_update_d0i3c_register(sdev, value);
+		if (ret < 0)
+			return ret;
 	} else {
 		/* prevent power gating in D0I0 */
 		flags = HDA_PM_PPG;
-	}
 
-	/* update D0I3C register */
-	ret = hda_dsp_update_d0i3c_register(sdev, value);
-	if (ret < 0)
-		return ret;
+		ret = hda_dsp_update_d0i3c_register(sdev, value);
```

Comment:
the change of indentation looks like a bug to me?

Reply:
it is contained by  "else {", so indentation is added.